mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -967,7 +970,7 @@ def _get_orderby_clauses(order_by_list, session):
                         # the column (is_nan) is not nullable. However it could become an issue
                         # if this precondition changes in the future.
                         (subquery.c.is_nan == sqlalchemy.true(), 1),
-                        (order_value.is_(None), 1),
+                        (order_value.is_(None), 2),
```

Comment:
Let's say we run code like this:

```python
with mlflow.start_run():
    log_metric("m", 1)

with mlflow.start_run():
    log_metric("m", np.nan)

with mlflow.start_run():
    pass

mlflow.search_runs(["0"], order_by=["metrics.m"])
```

This code creates a table looks like this, then sorts rows using the `metrics.m` and `clause_1` columns.

| run_id | metrics.m | is_nan | clause_1 |
| :----- | :-------- | :----- | :------- |
| 1      | 1         | False  | 0        |
| 2      | 0         | True   | 1        |
| 3      | NULL      | False  | 1        |

```sql
SELECT ... ORDER BY clause_1, metrics.m
```

In PostgreSQL, the result looks like this because NULLs are sorted AFTER non-null values:

| run_id | metrics.m | is_nan | clause_1 |
| :----- | :-------- | :----- | :------- |
| 1      | 1         | False  | 0        |
| 2      | 0         | True   | 1        |
| 3      | NULL      | False  | 1        |

In other databases (SQLite, MySQL, SQL server), the result looks like this because NULLs are sorted BEFORE non-null values:

| run_id | metrics.m | is_nan | clause_1 |
| :----- | :-------- | :----- | :------- |
| 1      | 1         | False  | 0        |
| 3      | NULL      | False  | 1        |
| 2      | 0         | True   | 1        |